appinventor/components/src/com/google/appinventor/components/runtime/Spreadsheet.java

```diff
@@ -661,6 +664,102 @@ public void FinishedWriteRow () {
+          BatchUpdateSpreadsheetRequest body = new BatchUpdateSpreadsheetRequest()
+              .setRequests(requests);
+          sheetsService.spreadsheets().batchUpdate(spreadsheetID, body).execute();
+          // Run the callback event block
+          activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+              FinishedAddSheet(sheetName);
+            }
+          });
+
+        } catch (Exception e) {
+          e.printStackTrace();
+          ErrorOccurred("AddSheet: " + e.getMessage());
+        }
+      }
+    });
+  }
+
+  @SimpleEvent(
+          description="The callback event for the addSheet block, called once the " +
+                  "values on the table have been updated.")
+  public void FinishedAddSheet(final String sheetName) {
+    EventDispatcher.dispatchEvent(this, "FinishedAddSheet", sheetName);
+  }
+
+  @SimpleFunction(description = "Deletes the specified sheet inside the Spreadsheet")
+  public void DeleteSheet(final String sheetName) {
+      if (spreadsheetID == null || spreadsheetID.isEmpty()) {
+          ErrorOccurred("DeleteSheet: " + "SpreadsheetID is empty.");
+          return;
+      } else if (credentialsPath == null || credentialsPath.isEmpty()) {
+          ErrorOccurred("DeleteSheet: " + "Credentials JSON is required.");
+          return;
+      }
+      // Run the API call asynchronously
+      AsynchUtil.runAsynchronously(new Runnable() {
+          @Override
+          public void run() {
+              try {
+                  Sheets sheetsService = getSheetsService();
+                  DeleteSheetRequest deleteSheetRequest = new DeleteSheetRequest()
+                      .setSheetId(getSheetID(sheetsService, sheetName)
+                      );
+                  List<Request> requests = new ArrayList<>();
+                  requests.add(new Request().setDeleteSheet(deleteSheetRequest));
+                  BatchUpdateSpreadsheetRequest body = new BatchUpdateSpreadsheetRequest()
+                      .setRequests(requests);
+                  sheetsService.spreadsheets().batchUpdate(spreadsheetID, body).execute();
+
+                  // Run the callback event block
+                  activity.runOnUiThread(new Runnable() {
+                      @Override
+                      public void run() {
+                          FinishedDeleteSheet(sheetName);
+                      }
+                  });
+
+              } catch (Exception e) {
+                  e.printStackTrace();
+                  ErrorOccurred("DeleteSheet: " + e.getMessage());
+              }
+          }
+      });
+  }
+
+  @SimpleEvent(
+          description="The callback event for the DeleteSheet block, called once the " +
+                  "values on the table have been updated.")
+  public void FinishedDeleteSheet(final String sheetName) {
+      EventDispatcher.dispatchEvent(this, "FinishedDeleteSheet", sheetName);
+  }
```

Comment:
The indentation here is not correct. Note that you can run `ant -Dcommit=ucr checkstyle` to check your changes and see what isn't correct.